EXT_PRAYTRACE.md

```diff
@@ -0,0 +1,258 @@
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `position/z`
+			- type: *(floatX)*
+			- description: *z-* coordinates
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+	
+	- `direction/'
+		- `direction/x`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `direction/y`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `direction/z`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+
+	- `eFieldSPolarisation/`
+		- `eFieldSPolarisation/x`
+			- type: *(complexX)*
+			- description: S polarisation along *x-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+		- `eFieldSPolarisation/y`
+			- type: *(complexX)*
+			- description: S polarisation along *y-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+		- `eFieldSPolarisation/z`
+			- type: *(complexX)*
+			- description: S polarisation along *z-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+	
+	- `eFieldPPolarisations/`
+		- `eFieldPPolarisations/x`
+			- type: *(complexX)*
+			- description: P polarisation along *x-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+		- `eFieldPPolarisations/y`
+			- type: *(complexX)*
+			- description: P polarisation along *y-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+		- `eFieldPPolarisations/z`
+			- type: *(complexX)*
+			- description: P polarisation along *z-* coordinate
+			- scope: *required*	
+			- unitSI: 1.0
+			- unitDimension: [sqrt(V / m) = sqrt(kg m s^-3 A^-1)] (1/2, 1/2, -3/2, -1/2, 0., 0., 0.)
+
+	- `photonEnergy`
+		- type: *(floatX)*
+		- description: energy, self-explanatory
+		- scope: *required*
+		- unitSI: 1.602176634e−19
+		- unitDimension: [J = kg m^2 s^-2] (2., 1., -2., 0., 0., 0., 0.)
+	
+	- `phase/`
+		- `phase/sPol`
+			- type: *(complexX)*
+			- description: phase for S polarisation
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+		- `phase/pPol`
+			- type: *(complexX)*
+			- description: phase for P polarisation
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+	
+### Additional Attributes for the Group `particlesPath`
```

Comment:
i would list all attributes together and indicate in the scope whether required or optional